src/sagas/pushNotification.js

```diff
@@ -0,0 +1,412 @@
+    return null;
+  }
+};
+
+/**
+ * localization utils to map the message key to the correct message to localize
+ */
+const localization = {
+  keys: new Set([
+    NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS,
+    NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS,
+    NEW_TRANSACTION_RECEIVED_TITLE
+  ]),
+  hasKey: (key) => localization.keys.has(key),
+  getMessage: (key, args) => {
+    if (!localization.hasKey(key)) {
+      console.log('unknown key', key);
+      return '';
+    }
+
+    let message = '';
+    if (key === NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITH_TOKENS) {
+      if (!args) {
+        console.log('args is null for key', key);
+        return '';
+      }
+      const countArgs = args.length;
+      if (countArgs === 3) {
+        const [firstToken, secondToken, other] = args;
+        const otherCount = parseInt(other, 10);
+        message = t`you have received ${firstToken}, ${secondToken} and ${otherCount} other token on a new transaction.`;
+      } else if (countArgs === 2) {
+        const [firstToken, secondToken] = args;
+        message = t`you have received ${firstToken} and ${secondToken}.`;
+      } else if (countArgs === 1) {
+        const [firstToken] = args;
+        message = `you have received ${firstToken}.`;
+      }
+    } else if (key === NEW_TRANSACTION_RECEIVED_DESCRIPTION_WITHOUT_TOKENS) {
+      message = t`There is a new transaction in your wallet.`;
+    } else if (key === NEW_TRANSACTION_RECEIVED_TITLE) {
+      message = t`New transaction received`;
+    }
+    return message;
+  }
+};
+
+const onForegroundMessage = async (message) => {
+  await messageHandler(message);
+};
+
+const onBackgroundMessage = async (message) => {
+  await messageHandler(message);
+};
+
+/**
+ * Handle the message received when application is in foreground and background (not closed) state
+ * @param {{ data: Object, from: string, messageId: string, sentTime: number, ttl: number }} message - Message received from wallet-service
+ * @example
+ * {
+ *   bodyLocArgs: '[\"10 T2\",\"5 T1\",\"2\"]',
+ *   bodyLocKey: 'new_transaction_received_description_with_tokens',
+ *   titleLocKey: 'new_transaction_received_title',
+ *   txId: 'txId1',
+ * }
+ * @inner
+ */
+const messageHandler = async (message) => {
+  const { data } = message;
+  if (!localization.hasKey(data.titleLocKey)) {
+    console.log('unknown message titleLocKey', data.titleLocKey);
+    return;
+  }
+
+  if (!localization.hasKey(data.bodyLocKey)) {
+    console.log('unknown message bodyLocKey', data.bodyLocKey);
+    return;
+  }
+
+  const bodyArgs = JSON.parse(data.bodyLocArgs);
+  const title = localization.getMessage(data.titleLocKey);
+  const body = localization.getMessage(data.bodyLocKey, bodyArgs);
+
+  notifee.displayNotification({
+    title,
+    body,
+    android: {
+      channelId: TRANSACTION_CHANNEL_ID,
+    },
+  });
+};
+
+const showPinScreenForResult = async (dispatch) => new Promise((resolve) => {
```

Comment:
Maybe export this as an util somewhere and re-use on `sagas/wallet.ts` and here instead of duplicating code

Reply:
I spent some time on the PushNotification refactoring. Can this refactor wait until I finish the main parts yet letting? I can do this refactoring in the next PRs.

Reply:
I just checked and this is exactly the same code as the one in `src/sagas/wallet.js`, wouldn't it be simple as exporting and using on both places?

I think it's bad to add duplicated code like this, we might forget to refactor it later

Reply:
I agree this should be reused, and it's better to be in this PR.